src/tty/mod.rs

```diff
@@ -30,10 +30,18 @@ mod windows;
 #[cfg(windows)]
 pub use self::windows::*;
 
+/// This struct contains information about the child process.
+pub struct ChildInfo {
+    #[cfg(unix)]
+    pub pid: libc::pid_t,
+}
+
 /// This trait defines the behaviour needed to read and/or write to a stream.
 /// It defines an abstraction over mio's interface in order to allow either one
 /// read/write object or a seperate read and write object.
 pub trait EventedReadWrite {
+    // TODO: should this maybe just be called `Pseudoterminal` or `Pty`?
+    // The current attempt at generality is probably unnecessary.
```

Comment:
If we dropped the pretense and had `trait Pseudoterminal` or similar, it'd feel less odd adding a `should_exit` method, which would avoid the gross `bool` on `process_signals`.

Reply:
I've also had issues with the name when this was initially introduced, but iirc there was a good reason for it and the naming was taken from a related API.

(See https://docs.rs/mio/0.6.16/mio/event/trait.Evented.html)